### DISCUSSION TITLE
Windows build 1 error

### DISCUSSION THREAD

**Discussion by krishnaavril** [AcademySoftwareFoundation/OpenRV#542] [Build Issues]
```       "C:\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "C:\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "C:\OpenRV\_build\cmake\dependencies\dependencies.vcxproj" (default target) (15) ->
       "C:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj" (default target) (164) ->
       (CustomBuild target) ->
         C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'C:\OpenRV\_build\CMakeFiles\2
       7ffb7829657fcab58aa9f832fd8f307\RV_DEPS_PYTHON3-requirements-flag.rule;C:\OpenRV\_build\CMakeFiles\27ffb7829657fcab58aa9f832fd8f307\RV_DEPS_PYSIDE2-build-flag.rule;C:\OpenRV\_build\C
       MakeFiles\6ef89c14370471f72f972cb264110d52\python310.dll.rule;C:\OpenRV\_build\CMakeFiles\dc0b82080efe119cccb8b58ef9885864\RV_DEPS_PYTHON3-stage-target.rule;C:\OpenRV\cmake\dependenc
       ies\CMakeLists.txt' exited with code 1. [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]

    4879 Warning(s)
    1 Error(s)

```

After various tests, I got one error, any thoughts on this please?

**Comment by bernie-laberge** [comment]
Hello @krishna8008 , 
sorry to hear that you're having difficulties building Open RV on windows.
Would you have the full log by chance ? We need to figure out what went wrong with the build of the RV_DEPS_PYTHON3 dependency. I typically search for "error:" in the log to find the root cause of a build break.

**Reply by krishnaavril** [reply]
Hello @bernie-laberge ,

Please have a look at the attatched log, Your help is appreciated,
I tried various methods and now I got the 2 errors, sorry!
[buildlog.log](https://github.com/user-attachments/files/16663472/buildlog.log)


I'm on Win10
Using VS 2022
Installed VS2019 as well

Commands:
cmake -B _build -G "Visual Studio 17 2022" -A x64 -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_WIN_PERL_ROOT=/c/Strawberry/perl/bin -DRV_DEPS_QT5_LOCATION=/c/Qt/5.15.2/msvc2019_64
cmake --build _build --config Release -v --parallel=32 --target main_executable

**Comment by bernie-laberge** [comment]
This is weird:
```
   126>CustomBuild:
         Traceback (most recent call last):
           File "C:\OpenRV\src\build\make_pyside.py", line 22, in <module>
             from utils import (
         ModuleNotFoundError: No module named 'utils'
```
Because the file utils.py is in the same directory as C:\OpenRV\src\build\make_pyside.py...

**Reply by bernie-laberge** [reply]
Perhaps we should be using :
`from .utils import (`
Would you mind doing a test on your end when you get the chance ?
At this line:
https://github.com/AcademySoftwareFoundation/OpenRV/blob/0c60e271cda40f46ee1eee64be735691ac3487aa/src/build/make_pyside.py#L22

**Reply by krishnaavril** [reply]
> Perhaps we should be using : `from .utils import (` Would you mind doing a test on your end when you get the chance ? At this line:
> 
> https://github.com/AcademySoftwareFoundation/OpenRV/blob/0c60e271cda40f46ee1eee64be735691ac3487aa/src/build/make_pyside.py#L22

[buildlog.log](https://github.com/user-attachments/files/16663762/buildlog.log)
Hello, @bernie-laberge I've changed the code as per your note,

Please check the updated full log attatched to this comment

```         Building PySide2 using C:/OpenRV/src/build/make_pyside.py
         Traceback (most recent call last):
           File "C:\OpenRV\src\build\make_pyside.py", line 22, in <module>
             from .utils import (
         ImportError: attempted relative import with no known parent package
   142>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'C:\OpenRV\_build\CMakeFiles\27ffb7829657fcab58aa9f832fd8f307\RV_DEPS_PYTHON3-requirements-flag.rule;C:\OpenRV\_build\CMakeFiles\27ffb7829657fcab58aa9f832fd8f307\RV_DEPS_PYSIDE2-build-flag.rule;C:\OpenRV\_build\CMakeFiles\6ef89c14370471f72f972cb264110d52\python310.dll.rule;C:\OpenRV\_build\CMakeFiles\dc0b82080efe119cccb8b58ef9885864\RV_DEPS_PYTHON3-stage-target.rule;C:\OpenRV\cmake\dependencies\CMakeLists.txt' exited with code 1. [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]
   142>Done Building Project "C:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj" (default targets) -- FAILED.
   ```

**Reply by krishnaavril** [reply]
After running various tests, latest msys2 is using a very latest python and some modules are not importing with it, we needed to install modules manually based on the build errors. can autodesk make it better and add a full compatible list for building. You already gave a best product just needed some makeup.

Thanks for the support
